mod_reforged/hooks/skills/perks/perk_mastery_cleaver.nut

```diff
@@ -0,0 +1,46 @@
+::Reforged.HooksMod.hook("scripts/skills/perks/perk_mastery_cleaver", function(q) {
+	q.onAdded = @(__original) function()
+	{
+		__original();
+		this.getContainer().add(::MSU.new("scripts/skills/perks/perk_rf_bloodlust", function(o) {
+			o.m.IsRefundable = false;
+			o.m.IsSerialized = false;
+		}));
+	}
+
+	q.onRemoved = @(__original) function()
+	{
+		__original();
+		this.getContainer().removeByID("perk.rf_bloodlust");
+	}
+
+	q.onTargetHit = @(__original) function( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
+	{
+		__original(_skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor);
+		if (_targetEntity.isAlive() && _damageInflictedHitpoints >= ::Const.Combat.MinDamageToApplyBleeding && !_targetEntity.getCurrentProperties().IsImmuneToBleeding && this.isSkillValid(_skill))
+		{
+			_targetEntity.getSkills().add(::new("scripts/skills/effects/bleeding_effect"));
+		}
+	}
+
+	q.onQueryTooltip = @(__original) function( _skill, _tooltip )
+	{
+		__original(_skill, _tooltip);
+
+		if (this.isSkillValid(_skill))
+		{
+			_tooltip.push({
+				id = _tooltip().top().id + 1,
```

Comment:
this should have a hard-coded id. I don't quite remember where you wanted onQueryTooltips to start though, was it 1000?

We might even want to use the same id for every skill. That way a submod can at least remove them all at once or search in them for a specific name.
A different id for every line is a very hard thing to guarantee even just within reforged as this has to be maintained over every different skill.